tests/acceptance/apps-deprovision-test.js

```diff
@@ -0,0 +1,108 @@
+import Ember from 'ember';
+import startApp from '../helpers/start-app';
+import { stubRequest } from '../helpers/fake-server';
+
+var App;
+
+module('Acceptance: App Deprovision', {
+  setup: function() {
+    App = startApp();
+  },
+  teardown: function() {
+    Ember.run(App, 'destroy');
+  }
+});
+
+test('/apps/:id/deprovision requires authentication', function(){
+  expectRequiresAuthentication('/apps/1/deprovision');
+});
+
+test('/apps/:id/deprovision will not deprovision without confirmation', function(){
+  var appId = 1;
+  var appName = 'foo-bar';
+
+  stubApp({
+    id: appId,
+    handle: appName
+  });
+
+  signInAndVisit('/apps/'+appId+'/deprovision');
+  andThen(function(){
+    var button = findWithAssert('button:contains(Deprovision)');
+    ok(button.is(':disabled'), 'deprovision button is disabled');
+  });
+});
+
+test('/apps/:id/deprovision will deprovision with confirmation', function(){
+  var appId = 1;
+  var appName = 'foo-bar';
+  var didDeprovision = false;
+
+  stubStack({id: '1'});
+
+  stubApp({
+    id: appId,
+    handle: appName,
+    _links: {
+      account: {href: '/accounts/1'}
+    }
+  });
+
+  stubRequest('post', '/apps/'+appId+'/operations', function(request){
+    didDeprovision = true;
+    return this.success({
+      id: '1',
+      status: 'queued',
+      type: 'deprovision'
+    });
+  });
+
+  stubStacks();
+
+  signInAndVisit('/apps/'+appId+'/deprovision');
+  fillIn('input[type=text]', appName);
+  andThen(function(){
+    $('input[type=text]').trigger('input');
```

Comment:
If the `button-with-confirmation`'s `isConfirmed` is a CP, this manual trigger appears to not be necessary.